_posts/2018-11-20-keeping-dependencies-updated.markdown

```diff
@@ -0,0 +1,138 @@
+configuration that'll let you know which packages will be updated before it ever creates a PR to update anything.
+If you click the edited dropdown on the PR body you'll see all the changes renovate made to the issue while I was
+trying to figure out the configuration.
+
+![GitHub PR edit history](/images/2018-11-20-keeping-dependencies-updated/issue-history.png)
+
+It took me a while to figure everything out, just take a look at the [commit history][pr-commits]. I'm going to
+work through the final setup just to give you an idea of our setup.
+
+First, I extended [Renovate][renovate]'s base config.
+
+```
+{
+  "extends": [
+    "config:base"
+  ],
+  ...
+}
+```
+
+If you've worked with [eslint][eslint], [babel][babel-extends], or other tools in the js ecosystem, you've probably
+seen this type of configuration extension. It essentially allows us to use their best practices out of the box.
+
+Next, I set the [assignees][assignees]. When [Renovate][renovate] opens a new PR, it'll assign it to these people
+so that the PR doesn't get missed.
+
+The actual meat of the change is the `packageRules` setup.
+
+```
+{
+  ...
+  "packageRules": [{
+    "packagePatterns": ["*"],
+    "excludePackagePatterns": ["^@artsy"],
+    "enabled": false
+  }],
+  ...
+}
+```
+
+[Renovate][renovate] allows you to set up multiple different `packageRules` and there's a lot of configuration for
+them. I'm not going to go through more than I did, but feel free to read more in their
+[docs][renovate-packagerules-docs]. In the `packageRule` that I setup, I specified `packagePatterns` with an
+asterisk to select all dependencies. Then using `excludePackagePatteners` I excluded anything that started with
+`@artsy`. Finally (and most importantly), I set `enabled` to `false` to disable the dependencies matching those
+combinations of rules. That last part took me a while to figure out. When you're building package rules in
+[Renovate][renovate], think of it as building out a list of operations to perform.
+
+The last few pieces of config are a little more straight-forward and you can read about those in the docs. The one
+thing that I'll mention is that [vulnerabilityAlerts][renovate-vulnerabilityalerts] _ignores_ `packageRules` and
+triggers update PRs for anything that's reported to have a vulnerability. You'll have to explicitly disable it if
+you only want reports on certain packages. Though, having it on probably isn't a bad idea...
+
+### Wrapping up
+
+So, that's how we configured [Renovate][renovate] to automatically update npm dependencies in Artsy's namespace.
+It's been extremely useful already. I also went ahead and pulled our configuration out into a [shared
+repo][artsy-renovate-config] so that we didn't have to copy these configurations across all of our projects. That's
+a blog post for another day.
+
+Be well friends.
+
+[design-system]:
+  https://www.uxpin.com/studio/blog/design-systems-vs-pattern-libraries-vs-style-guides-whats-difference/
```

Comment:
you may need to use a prettier ignore for this line, I *think* it can result in broken links when on the website